src/lib/lmdb_storage/generic.ml

```diff
@@ -328,8 +328,183 @@ let%test_module "Lmdb storage tests" =
           cnt := !cnt + 1 ;
           `Continue ) ;
       assert (!cnt = !odd_cnt) ;
+
+      Deferred.unit
+
+    let%test_unit "Ro: Get and Iter operation within with_txn scope" =
+      let n = 300 in
+      let hm = Hashtbl.create (module Int) in
+      test_with_dir
+      @@ fun dir ->
+      let env, db = Rw.create dir in
+      Quickcheck.test
+        (Quickcheck.Generator.both uint32
+           (String.gen_with_length 100 Base_quickcheck.quickcheck_generator_char) )
+        ~trials:n
+        ~f:(fun (k, v_str) ->
+          let v = Bigstring.of_string v_str in
+          Hashtbl.add ~key:k ~data:v hm
+          |> function `Duplicate -> () | `Ok -> Rw.set ~env db k v ) ;
+
+      let ro_env, ro_db = Ro.create dir in
+      let hm_from_iter = Hashtbl.create (module Int) in
+      let (_ : Bigstring.t option) =
```

Comment:
Could you explain the meaning of different result here? It's just ignored here. I digged down a bit and didn't find docs in our codebsae.

Reply:
![image](https://github.com/user-attachments/assets/22d7ee60-99ce-4e41-a86a-de6ffe211408)


We should unwrap this option. If this it's None, this should be a test failure.

Reply:
See [here](https://drup.github.io/ocaml-lmdb/dev/lmdb/Lmdb/Txn/index.html#val-go).